src/main/java/com/r307/arbitrader/service/TradingService.java

```diff
@@ -468,10 +469,19 @@ private void exitPosition(Spread spread) {
     }
 
     // if feeComputation == CLIENT we want to compute the fees and add them to the volume
-    private BigDecimal addFees(Exchange exchange, CurrencyPair currencyPair, BigDecimal volume) {
+    private BigDecimal addFees(Exchange exchange, CurrencyPair currencyPair, BigDecimal volume, boolean isShortFee) {
         if (exchangeService.getExchangeMetadata(exchange).getFeeComputation().equals(FeeComputation.CLIENT)) {
+            final ExchangeFee exchangeFee = exchangeService.getExchangeFee(exchange, currencyPair, true);
+
+            if (isShortFee && !exchangeFee.getShortFee().isPresent()) {
+                LOGGER.error("exchange:{}|missing margin fee for this exchang. Go to your application.yml and set a margin fee for this exchange",
+                    exchange.getExchangeSpecification().getExchangeName());
+                // Crash the bot
+                throw new RuntimeException("Missing margin fee configuration for exchange: " + exchange.getExchangeSpecification().getExchangeName());
+            }
+
             BigDecimal fee = volume
-                .multiply(exchangeService.getExchangeFee(exchange, currencyPair, true))
+                .multiply(isShortFee ? exchangeFee.getShortFee().get() : exchangeFee.getLongFee())
```

Comment:
This will choose either the long or the short fee. What we want here, and in `subtractFees()`, and in the regular `feeComputation == SERVER` flow, is to choose whether we use the trade fee alone or the trade fee plus the margin fee. Right now that's based on long or short but it's important to remember that we're making a big assumption there. Short trades _must_ be on margin, but even though it's not possible (yet) in Arbitrader it is possible to make a long trade on margin. That's why I want the names to be right.